packages/features/bookings/lib/handleNewBooking.ts

```diff
@@ -586,29 +588,55 @@ async function handler(
   const contactOwnerEmail = skipContactOwner ? null : contactOwnerFromReq;
 
   let routingFormResponse = null;
-
   if (routedTeamMemberIds) {
     //routingFormResponseId could be 0 for dry run. So, we just avoid undefined value
     if (routingFormResponseId === undefined) {
       throw new HttpError({ statusCode: 400, message: "Missing routingFormResponseId" });
     }
-    routingFormResponse = await prisma.app_RoutingForms_FormResponse.findUnique({
-      where: {
-        id: routingFormResponseId,
-      },
-      select: {
-        response: true,
-        form: {
-          select: {
-            routes: true,
-            fields: true,
+
+    if (queuedFormResponse) {
+      // Write to the routing form response table and return the result
+      routingFormResponse = await RoutingFormResponseRepository.writeQueuedFormResponseToFormResponse(
+        routingFormResponseId,
+        bookerEmail
+      );
+      if (!routingFormResponse) {
+        throw new HttpError({ statusCode: 400, message: "Missing routingFormResponse" });
+      }
+      routingFormResponseId = routingFormResponse.id;
```

Comment:
Reassigning the `routingFormResponseId` to point to the new record in the response table.

I decided to take the approach of writing the record here because `handleNewBooking` expects the routing form responses and to point to the record in the response table.